### DISCUSSION TITLE
[Question] Script to output airfoil/wing shapes

### DISCUSSION THREAD

**Discussion by r-ashwin** [su2code/SU2#1235] [Q&A]
(new user here and apologies if this has been raised/addressed before).

I want to perturb the Hicks and Henne bump function design variables and write out the airfoil shape, __without running the primal or adjoint solvers__. So I want to implement `Y = f(X)` where `X` is the vector of Hicks and Henne bump function weights and `Y` is a vector of the actual `y` coordinate of the airfoil at each `x` corresponding to an element of `X`.

Is there a script that already exists to this effect? If so can you share that with me? If not do you have suggestions as to how to implement this? Thank you!

**Comment by pcarruscag** [comment]
Run SU2_DEF

**Reply by r-ashwin** [reply]
Thank you. When I run SU2_DEF, it prints out two files: `boundary.dat` and `mesh_out.su2`. The first file contains--what looks like--mesh element information for the unstructured grid for the boundaries of the domain. The second file seems to include similar information in addition to the `(x,y)` coordinates of all the nodes in the entire mesh.

Is there an existing way I can output just the airfoil `(x,y)` coordinates as a file? If not, can you please explain the organization of `boundary.dat` and its relation to the `mesh_out.su2` so I can extract this data?

PS: please feel free to point me to the right part of the documentation that addresses this.

**Reply by pcarruscag** [reply]
Add something like this to your config:
OUTPUT_FILES= SURFACE_CSV, SURFACE_PARAVIEW, SURFACE_PARAVIEW_ASCII
The CSV will be a list of boundary points for the surfaces in MARKER_PLOTTING.
If you know the order in which the boundary points are written to the mesh file you just sort the CSV accordingly and that is your deformed airfoil.
If you do not have that information, the PARAVIEW_ASCII contains the definition of the elements (vtk format).
The SU2 mesh format is this one https://su2code.github.io/docs/Mesh-File/
The boundary file is essentially just the boundary part, but it is a kind of temporary, for what you are trying to do the other files are better.

Maybe there is some functionality in SU2_GEO to automate all this but I do not know that part of SU2 so well.
